nixops/backends/__init__.py

```diff
@@ -7,52 +7,26 @@
 import nixops.util
 import nixops.resources
 import nixops.ssh_util
-import xml.etree.ElementTree as ET
 
 
 class MachineDefinition(nixops.resources.ResourceDefinition):
     """Base class for NixOps machine definitions."""
 
-    def __init__(self, xml, config={}) -> None:
```

Comment:
What is the difference here between `config` and `xml`? A literal translation might be `s/xml/json/` and keeping `config`,  what is in `config` which we're giving up?

Reply:
`xml` is an instance of `xml.etree.ElementTree.Element` 

`config` was a previous attempt at making Python objects from the eval rather than working with the XML tree in every resource using this method https://github.com/NixOS/nixops/blob/ba0c3f6/nixops/util.py#L449.

They both contain the same data, just in different formats. It seems to me like the `config` parameter never really got any usage anywhere.